[ListItem] add text prop

cgestes opened the pull request:
see #1072

hai-cea commented:
@cgestes Sorry - I just merged in the nested list item PR (#918). It will cause conflicts.

hai-cea commented:
@cgestes Please remember to update the new MenuItem component as well. Also, `primaryText` can be a string or an element just like `secondaryText`.

cgestes commented:
rebased.

Fixed NestedItem at the same time. (everything that is a ReactElement is considered a NestedItem).

What do you think about setting proptypes of primaryText to required?

hai-cea commented:
@cgestes We may not want to make it required since `MenuItem` uses `ListItem` and I can see a case for having a blank menu item for optional selects.

hai-cea commented:
@cgestes Are you planning on implementing the changes in `MenuItem` as well?

cgestes commented:
ups wrote a comment about it, but did not press send.

Right now the changeset works without touching MenuItem.

I wonder what to do with MenuItem. can we just forward primaryText and secondaryText and consider all childs to be subitems/submenus?

hai-cea commented:
@cgestes Yes, that's what I was thinking. MenuItems will have a `primaryText` prop as well which will allow for nested MenuItems. Do you think it's necessary?

hai-cea commented:
Thanks @cgestes

cgestes commented:
Yes I started implementing MenuItem like that.

cgestes commented:
You did it.. Thank you.

Sorry I did not have many time for mui today.

hai-cea commented:
@cgestes No worries - Your PR worked just fine. I was just cleaning the docs and it involved a lot of mindless typing. :)
